numba/tests/test_listimpl.py

```diff
@@ -421,3 +458,39 @@ def test_sizing(self):
         # Check different sizes of the key & value.
         for i in range(1, 16):
             self.check_sizing(item_size=i, nmax=2**i)
+
+    def test_mutability(self):
+        # setup and populate a singleton
+        l = List(self, 8, 1)
+        one = struct.pack("q", 1)
+        l.append(one)
+        self.assertTrue(l.is_mutable)
+        self.assertEqual(len(l), 1)
+        r = struct.unpack("q", l[0])[0]
+        self.assertEqual(r, 1)
+
+        # set to immutable and test guards
+        l.set_immutable()
+        self.assertFalse(l.is_mutable)
+        # append
+        with self.assertRaises(ValueError):
```

Comment:
Can you make this check for the error message? `ValueError` is common. This can be caused by other reasons.

Reply:
done